server/src/services/commits.ts

```diff
@@ -113,6 +115,33 @@ const addCommit = async (
   );
 };
 
+/**
+ * Pays for the commit with the specified commitId.
+ * An error will be thrown if the commit is not SUCCESSFUL.
+ * @param commitId Id of the commit to be deleted
+ * @param paymentData Data of the payment request
+ */
+const payForCommit = async (
+  commitId: number,
+  paymentData: CommitPaymentRequest
+) => {
+  // Check that listing exists
+  const commit = await commitStorage.getCommit(commitId);
+
+  // Check that listing is successful
+  if (commit.commitStatus !== 'successful') {
```

Comment:
Did we discuss in previous project meeting that we allow merchants to still sell the product despite not enough customers are committing (and hence not successful)?

Reply:
@xlai20 That's an unsuccessful commit right? I don't think merchants are allowed to sell that, it was the hard requirement we agreed on during meeting last time.